extensions/coroutines-extensions/src/commonMain/kotlin/com/squareup/sqldelight/runtime/coroutines/FlowExtensions.kt

```diff
@@ -20,81 +20,69 @@ package com.squareup.sqldelight.runtime.coroutines
 
 import com.squareup.sqldelight.Query
 import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.FlowPreview
-import kotlinx.coroutines.channels.Channel.Factory.CONFLATED
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.flowViaChannel
-import kotlinx.coroutines.flow.flowWith
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.channelFlow
+import kotlinx.coroutines.flow.conflate
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.mapNotNull
+import kotlinx.coroutines.withContext
 import kotlin.coroutines.CoroutineContext
 import kotlin.jvm.JvmName
 import kotlin.jvm.JvmOverloads
 
 /** Turns this [Query] into a [Flow] which emits whenever the underlying result set changes. */
-@FlowPreview
+@ExperimentalCoroutinesApi
 @JvmName("toFlow")
-fun <T : Any> Query<T>.asFlow(): Flow<Query<T>> = flowViaChannel(CONFLATED) { channel ->
+fun <T : Any> Query<T>.asFlow(): Flow<Query<T>> = callbackFlow<Query<T>> {
   val listener = object : Query.Listener {
     override fun queryResultsChanged() {
-      channel.offer(this@asFlow)
+      offer(this@asFlow)
     }
   }
   addListener(listener)
-  @Suppress("EXPERIMENTAL_API_USAGE") //TODO: Remove when invokeOnClose is no longer experimental, or use replacement.
-  channel.invokeOnClose {
+  awaitClose {
     removeListener(listener)
   }
-  channel.offer(this@asFlow)
-}
+  offer(this@asFlow)
+}.conflate()
```

Comment:
Why? This seems like it should be left to the user

Reply:
Two reasons:
1. This is the how you keep the behavior of the previous code which was using `Channel.CONFLATED`.
2. Not conflating it would lead to unexpected behavior as we are using `offer`, and newer values would be ignored, and the query would not be run again if it is updated while collecting a value.

Reply:
The previous implementation conflated the channel being used to send the events.. This creates an unnecessary intermediate channel. Pass conflated to `callbackFlow` instead to retain the old behavior.

Reply:
That's not how it works now. `callbackFlow` does not take a capacity parameter.
`conflate()` is an alias to the `buffer`  operator called with `Channel.CONFLATED`, and the adjacent call to the `buffer` operator has an effect on the resulting flow, which makes the (single) channel from `callbackFlow`/`channelFlow` use the capacity passed to the buffer operator. You can see for yourself by looking at the sources from kotlinx.coroutines 1.3.0-M1.

Reply:
Ok let's switch to just using `flow` directly then. Actually, let's do `unsafeFlow` since we can guarantee our correctness given the channel usage.

Reply:
We can't use `flow`here because we're dealing with callbacks, not suspending functions. `callbackFlow` (or the equivalent `channelFlow`), plus usage of the `buffer` operator (`conflate()` in that particular case) to configure the underlying `Channel` is what we need.

FYI, `unsafeFlow` is internal to kotlinx.coroutines.

Reply:
`flow` works fine. You just create the channel yourself.

Reply:
Why do so?
`callbackFlow` + `conflate` works right, and allows code reuse.

Reply:
`conflate` does some nasty shit internally to mutate the upstream channel. I'd rather just avoid it.

Reply:
Well, it changes the capacity of the only upstream channel, before returning the ready-to-use `Flow`. That means the `Flow` should support freezing for Kotlin/Native.

Whether it's "nasty shit" or not is not our problem as long as it remains efficient. Remember this is not API surface, only implementation details. I'm just using the public experimental API as it has been designed.

Also, I don't think it's right to make the `asFlow` implementation more complex because of its implementation details, that could change anyway.

I you think `callbackFlow` + `conflate` is not a good API design, or is improperly implemented, I encourage you to [open an issue on kotlinx.coroutines repo](https://github.com/Kotlin/kotlinx.coroutines/issues/new) to discuss it, and maybe suggest an alternative. This will probably received consideration as the API is currently experimental. We can leave this PR open until it is resolved if needed.

Reply:
Also, I just checked, and there's no mutation taking place into the `buffer` operator.
Instead, it creates the actual `Channel from the `ChannelFlow` with the capacity (from the internal `update` function`, and returns a `Flow` of it. In case the `Flow` it is applied on is not a `ChannelFlow`, then it uses another mechanism to buffer other `Flow`s, but that one is irrelevant for our case here.

Reply:
I'll try to find time to distill a thought out comment, but the likelihood is small.